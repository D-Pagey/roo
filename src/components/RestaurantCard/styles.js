import styled from 'styled-components';

export const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    max-width: 354px;
`;

export const List = styled.ul`
    color: #828585;
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;
`;
