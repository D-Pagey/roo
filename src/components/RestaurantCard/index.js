import React from 'react';
import { arrayOf, number, string } from 'prop-types';
import * as S from './styles';

const RestaurantCard = ({ imageURL, link, name, price, tags }) => (
    <S.Link href={link} target="_blank" rel="noopener noreferrer">
        <S.Image src={imageURL} alt={name} />

        <S.Title>{name}</S.Title>
        <S.List>
            {tags.map((tag) => (
                <React.Fragment key={tag}>
                    <li>{tag}</li>
                    <S.Bullet>·</S.Bullet>
                </React.Fragment>
            ))}
            {'£'.repeat(price)}
        </S.List>
    </S.Link>
);

RestaurantCard.propTypes = {
    imageURL: string.isRequired,
    link: string.isRequired,
    name: string.isRequired,
    price: number.isRequired,
    tags: arrayOf(string).isRequired
};

export default RestaurantCard;
