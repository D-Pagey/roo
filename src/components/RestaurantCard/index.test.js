import React from 'react';
import { render } from '@testing-library/react';
import RestaurantCard from '.';

const props = {
    imageURL:
        'https://f.roocdn.com/images/menus/25785/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1477308353',
    link: 'https://deliveroo.co.uk/menu/london/bank/burrito-kitchen-cheapside',
    name: 'Burrito Kitchen',
    price: 2,
    tags: ['Mexican', 'Burritos', 'Halal']
};

describe('RestaurantCard component', () => {
    it('should render', () => {
        const { container } = render(<RestaurantCard {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
