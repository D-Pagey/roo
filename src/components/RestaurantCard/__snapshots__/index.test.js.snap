// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard component should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 354px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c1 {
  height: auto;
  margin: 0 0 0.5rem;
  max-width: 100%;
}

.c2 {
  color: #2e3333;
  font-size: 1.375rem;
  font-weight: 500;
  margin: 0 0 5px;
}

.c3 {
  color: #828585;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c4 {
  margin: 0 0.5rem;
}

<a
  class="c0"
  href="https://deliveroo.co.uk/menu/london/bank/burrito-kitchen-cheapside"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Burrito Kitchen"
    class="c1"
    src="https://f.roocdn.com/images/menus/25785/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1477308353"
  />
  <h5
    class="c2"
  >
    Burrito Kitchen
  </h5>
  <ul
    class="c3"
  >
    <li>
      Mexican
    </li>
    <span
      class="c4"
    >
      ·
    </span>
    <li>
      Burritos
    </li>
    <span
      class="c4"
    >
      ·
    </span>
    <li>
      Halal
    </li>
    <span
      class="c4"
    >
      ·
    </span>
    ££
  </ul>
</a>
`;
